tests/test_coordinate_source.py

```diff
@@ -392,13 +392,11 @@ def testImages(self):
             self.assertTrue(np.array_equal(imgs_coord[i], imgs_star[i]))
 
     def testCached(self):
-        src_cached = BoxesCoordinateSource(self.files_box)
-        src_uncached = BoxesCoordinateSource(self.files_box)
-        src_cached.cache()
+        src = BoxesCoordinateSource(self.files_box)
+        src_uncached = src[:]
+        src.cache()
```

Comment:
Unless I have missed something, cache should i) generate images ii) reset only the generation_pipeline belonging to this source.  No metadata or image data should be mutated afaik.

It is true the pipeline reset is a mutation, but we will be recalling the array backed images, and so there should be no net effect to anything outside of `.images`.

It is probably true that in the immutable framework we no longer need the `self.generation_pipeline_forward(...)` line when loading ImageSource._cached_im images.  Anything adding to the pipeline after a cache call should yield a new source.  I can rm that or create an issue (if we continue this path)

(Another path.) Vineet had converted most of the cache calls to the immutable style for us and I reverted those because I didn't feel like caching was a transform.  If you are worried about it, or feel differently, I can quickly revert reverting the cache calls, so, making them to the immutable syntax. I think its just one commit and a couple missed lines from merges.

Let me know which way you prefer and I'll get it done. Thanks

Reply:
Agreed that caching is not a transform, but it might still be confusing that all other operations on sources yield a new source but this one does not. I think it's a simpler rule of thumb to have all operations (whether they mutate or not, strictly speaking) generate a new source. This is a cached *view* of the source, if you will.

Reply:
No worries, I'll put that back, just a few minutes.

Reply:
Reverted. My bad on that one.

Reply:
Totally fine. I see what you were trying to do. Just think it's simpler this way.